source/extensions/config_subscription/grpc/new_grpc_mux_impl.cc

```diff
@@ -71,8 +71,26 @@ NewGrpcMuxImpl::createGrpcStreamObject(GrpcMuxContext& grpc_mux_context) {
               grpc_mux_context.rate_limit_settings_);
         },
         /*failover_stream_creator=*/
-        // TODO(adisuissa): implement when failover is fully plumbed.
-        absl::nullopt,
+        grpc_mux_context.failover_async_client_
+            ? absl::make_optional(
+                  [&grpc_mux_context](
+                      GrpcStreamCallbacks<envoy::service::discovery::v3::DeltaDiscoveryResponse>*
+                          callbacks)
+                      -> GrpcStreamInterfacePtr<
+                          envoy::service::discovery::v3::DeltaDiscoveryRequest,
+                          envoy::service::discovery::v3::DeltaDiscoveryResponse> {
+                    return std::make_unique<
+                        GrpcStream<envoy::service::discovery::v3::DeltaDiscoveryRequest,
+                                   envoy::service::discovery::v3::DeltaDiscoveryResponse>>(
+                        callbacks, std::move(grpc_mux_context.failover_async_client_),
+                        grpc_mux_context.service_method_, grpc_mux_context.dispatcher_,
+                        grpc_mux_context.scope_,
+                        // TODO(adisuissa): the backoff strategy for the failover should
+                        // be the same as the primary source.
+                        std::make_unique<FixedBackOffStrategy>(500),
```

Comment:
nit could factor out this magic number `500` into constant

Reply:
Done, although IMHO it makes the code less readable (because of the template).